### PR TITLE
docs update: dev intros and further "rapid learning curve"

### PR DESCRIPTION
**Pull request by atlas0fd00m**

### CODE REVIEW

File: docs/vivisect/devguides/gettingstarted.rst [hunk 1]
```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
...
+
+    In [65]: vw.analyze() 
+
+    In [66]: vw.setMeta('StorageName', '/home/atlas/work/chown-new.viv') 
+
+    In [67]: vw.saveWorkspace(True) 
+
+`VivWorkspace.getFunctions()` returns a list of Virtual Addresses (va's) for the beginning of each function::
+
+    In [67]: vw.getFunctions()
```
Comment (rakuy0) [inline docs/vivisect/devguides/gettingstarted.rst:212]: As a formatting note, it's a little jarring and messy to mix raw python with the ipython markups. I'd like to standardize on one. I'm not particularly picky about which, but I'd like to be consistent.

**Review by rakuy0 — CHANGES_REQUESTED** [review]
Looking really good. I've got a lot of little tweaks/wordsmithing requests, but overall I like it.

How do the docs look if you build them and look at the files in your browser?

File: docs/vivisect/devguides/gettingstarted.rst [hunk 2]
```diff
@@ -3,10 +3,40 @@
...
+    
+    * VDB (vdb/) - Higher level Vulnerability Debugger code (depends on VTrace)
+    
+    * VStruct (vstruct/) - Easy-to-use Structure builder/manipulation library
+    
+    * VisGraph (visgraph/) - Graph and Path libraries used to represent virtually anything with nodes and edges with properties
+    
+    * Vivisect (vivisect/) - Rich multi-architecture, cross-platform binary analysis framework, depending on most of the other libraries
+
+    * Symboliks (vivisect/symboliks/) - Symbolic analysis framework based on Vivisect
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:28]: added.

File: docs/vivisect/devguides/gettingstarted.rst [hunk 3]
```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
...
+
+    In [36]: vw.getMeta('StorageName')                                                                                                 
+    Out[36]: '/home/atlas/work/firmware.hex.viv'
+
+    In [37]: vw.setMeta('StorageName', '/home/atlas/work/firmware.hex-clean-211205.viv')
+
+    In [38]: vw.saveWorkspace(fullsave=True)                                                                                           
+
+
+Loading an ELF and Working With Functions
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:183]: modified

File: docs/vivisect/devguides/gettingstarted.rst [hunk 4]
```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
...
+    # Now we create a VivWorkspace and load in the binary
     import vivisect
     vw = vivisect.VivWorkspace()
     vw.loadParsedBin(binary_object)
     vw.analyze()
 
 where `binary_object` is your PE or Elf class object.
+
+
+My Comfort-Zone
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:87]: i'm ok with that, but i'm not sure i want to take away the ability to first-person the next lines tho.  
this needs to seem easy and simple and happy... i think that first-person gets that done better.

File: docs/vivisect/devguides/gettingstarted.rst [hunk 5]
```diff
@@ -3,10 +3,40 @@
...
+    
+    * Cobra (cobra/) - Networking abstraction library, including remote-code-pulling and clustering facilities
+    
+    * VQT (vqt/) - Core Vivisect QT windowing libraries (great for extending Vivisect or writing your own code)
+    
+    * VTrace (vtrace/) - Platform-agnostic low-level debugging/tracing library
+    
+    * VDB (vdb/) - Higher level Vulnerability Debugger code (depends on VTrace)
+    
+    * VStruct (vstruct/) - Easy-to-use Structure builder/manipulation library
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:22]: how so?  `vivisect.parsers`?

File: docs/vivisect/devguides/gettingstarted.rst [hunk 6]
```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
...
+    binary_object = Elf.Elf(open("/path/to/binary", "rb"))
+
+    # Now we create a VivWorkspace and load in the binary
     import vivisect
     vw = vivisect.VivWorkspace()
     vw.loadParsedBin(binary_object)
     vw.analyze()
 
 where `binary_object` is your PE or Elf class object.
+
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:85]: done

File: docs/vivisect/devguides/gettingstarted.rst [hunk 7]
```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
...
+    In [12]: print(vw.config.reprConfigPaths())                                                                                        
+    Valid Config Entries:
+        remote.server = 10.42.120.72
+        vdb.BreakOnEntry = False
+        vdb.BreakOnMain = False
+        vdb.SymbolCacheActive = True
+        ...
+        viv.parsers.srec.arch = rxv2
+        viv.parsers.srec.offset = 192
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:129]: yep.  so is RXv2 et al.  lol.  lots of cool work we have going.

File: docs/vivisect/devguides/gettingstarted.rst [hunk 8]
```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
...
+
+    In [32]: vw.analyze()
+
+When you're happy with your workspace, be sure to save it::
+
+    In [33]: vw.saveWorkspace(fullsave=True)
+
+To save a workspace, `vw.saveWorkspace()` is used.  The "fullsave=True" means to write a complete file, instead of saving incrementally.  For the first time save, this is important, as it places the header on the workspace file which tells Viv what kind of file it is.
+
+No filename is given, but the filename to be written is stored in the workspace metadata.  The default name is the last file loaded into the workspace + ".viv".  You can see and modify this filename like so::
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:173]: ??  where do we specify a filename?

[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:173]: i'd like to talk about modifying the api to *allow* setting a filename easily, without having to manually modify `StorageName`

File: docs/vivisect/devguides/gettingstarted.rst (hunk 1 above)
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:212]: i understand the desire for consistency.  i'm not sure i agree that they should be treated the same, especially in this format (markdown).  
ipython gives context and a real feeling like you're "in the terminal with the author" whereas the others are easily copy/pasted.

i can be convinced... but i'm not certain that i am yet.  

thoughts?

File: docs/vivisect/devguides/gettingstarted.rst [hunk 9]
```diff
@@ -36,14 +66,287 @@ If you've already read a file into memory, you can instead use the `loadFromFd`
...
+     ('e705a3d616747893990567e06257be59',
+      0x200b570,
+      0x200b586,
+      {'va1': 0x200b584, 'va2': 0x200b586, 'codeflow': (0x200b584, 0x200b586)})]
+    
+
+Loading a PE and Listing Its Exports
+====================================
+
+Having More Fun
```
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:339]: heh.  when i'm done.  for now, i may cut it and add it back later (or in another module).

i'd like to do some intro to symboliks and intro to emulation code there.

[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:339]: oh, for that section, that's easy.

File: docs/vivisect/devguides/gettingstarted.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (rakuy0) [reply docs/vivisect/devguides/gettingstarted.rst:22]: No, just the `vstruct.vsParse` stuff, though on a second read, you might have put that with the builder stuff, but I feel like it deserves it's own call out.

File: docs/vivisect/devguides/gettingstarted.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (rakuy0) [reply docs/vivisect/devguides/gettingstarted.rst:173]: No, I mean a wording tweak. You have `No filename is given`, which we can make a little bit better in terms of grammar.

And we can specify a new filename via the StorageName meta, but a new param might not be amiss,

File: docs/vivisect/devguides/gettingstarted.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:22]: yeah, i fit "parsing" into builder and manipulation, but i can add parsing in there.

File: docs/vivisect/devguides/gettingstarted.rst (hunk 8 above)
[the remark this answers is not in the document]
Comment (atlas0fd00m) [reply docs/vivisect/devguides/gettingstarted.rst:173]: gotcha.

**Comment by atlas0fd00m** [comment]
i'm not sure how to build the docs on my machine.  i've been using a md-editor called ReText.

**Review by rakuy0 — APPROVED** [review]

**Comment by rakuy0** [comment]
Oh, for later reference (and I'll throw this in it's own docs), you can cd in the `docs` directory and run `make html`, and the open the file `_build/html/index.html` in your browser to see your built docs.

**Pull request merged**